app/components/channel_drawer/channel_list.js

```diff
@@ -143,24 +148,131 @@ class ChannelList extends React.Component {
     };
 
     handleClose = (channel) => {
+        this.setState({showOptions: false});
+        this.props.handleCloseDM(channel);
+    };
+
+    handleLeave = (channel, term) => {
         const {formatMessage} = this.props.intl;
         Alert.alert(
-            formatMessage({id: 'mobile.channel_list.alertTitleCloseDM', defaultMessage: 'Close Direct Message'}),
+            formatMessage({id: 'mobile.channel_list.alertTitleLeaveChannel', defaultMessage: 'Leave {term}'}, {term}),
             formatMessage({
-                id: 'mobile.channel_list.alertMessageCloseDM',
-                defaultMessage: 'Are you sure you want to close the DM with {username}'
+                id: 'mobile.channel_list.alertMessageLeaveChannel',
+                defaultMessage: 'Are you sure you want to leave the {term} with {name}?'
             }, {
-                username: channel.display_name
+                term: term.toLowerCase(),
+                name: channel.display_name
             }),
-            [
-                {text: formatMessage({id: 'mobile.channel_list.alertNo', defaultMessage: 'No'})},
-                {
-                    text: formatMessage({id: 'mobile.channel_list.alertYes', defaultMessage: 'Yes'}),
-                    onPress: () => this.props.handleCloseDM(channel)
-                }]
+            [{
+                text: formatMessage({id: 'mobile.channel_list.alertNo', defaultMessage: 'No'})
+            }, {
+                text: formatMessage({id: 'mobile.channel_list.alertYes', defaultMessage: 'Yes'}),
+                onPress: () => {
+                    this.setState({showOptions: false});
+                    this.props.handleLeaveChannel(channel);
+                }
+            }]
         );
     };
 
+    onShowModal = (channel) => {
+        const {formatMessage} = this.props.intl;
+        let open;
+        let close;
+        let favorite;
+        let modalTitle;
+
+        this.props.handleDisableDrawer(true);
```

Comment:
Maybe make this automatically happen when a `ModalOptions` is displayed? I can't think of a case where we'd want the drawers available in the modals